src/validation.cpp

```diff
@@ -3018,6 +3018,7 @@ bool Chainstate::ActivateBestChain(BlockValidationState& state, std::shared_ptr<
                     pindexMostWork = nullptr;
                 }
                 pindexNewTip = m_chain.Tip();
+                m_chainman.m_best_header = pindexNewTip;
```

Comment:
Pretty sure that this isn't correct, the headers-only chain is synchronized first. So, during IBD, it will be ahead of the "tip" block that we are connecting, and you are going to overwrite it.